File: ConexaoVagas Front/src/models/vaga.ts
import { BeneficioVaga } from "./beneficioVaga";
import { Candidatura } from "./candidatura";
import { Empresa } from "./empresa";
import { Endereco } from "./endereco";
import { HabilidadeVaga } from "./habilidadeVaga";
import { Matching } from "./matching";

export class Vaga {
    idVaga?: number;
    idEmpresa?: number;
    idEndereco?: number | null;
    titulo?: string;
    salario?: number;
    qualificacao?: string;
    dataCriado?: string;
    dataExpiracao?: string;
    cargaHoraria?: number;
    descricao?: string;
    visualizacao?: number;
    idEmpresaNavigation?: Empresa = undefined;
    idEnderecoNavigation?: Endereco = undefined;
    beneficioVaga?: BeneficioVaga[] = [];
    candidatura?: Candidatura[] = [];
    habilidadeVaga?: HabilidadeVaga[] = [];
    matching?: Matching[] = [];
}

File: ConexaoVagas API/Domains/Vaga.cs
using System;
using System.Collections.Generic;

namespace ConexaoVagasAPI.Domains
{
    public partial class Vaga
    {
        public Vaga()
        {
            BeneficioVaga = new HashSet<BeneficioVaga>();
            Candidatura = new HashSet<Candidatura>();
            HabilidadeVaga = new HashSet<HabilidadeVaga>();
            Matching = new HashSet<Matching>();
        }

        public int IdVaga { get; set; }
        public int IdEmpresa { get; set; }
        public int? IdEndereco { get; set; }
        public string Titulo { get; set; }
        public decimal Salario { get; set; }
        public string Qualificacao { get; set; }
        public DateTime DataCriado { get; set; }
        public DateTime DataExpiracao { get; set; }
        public int CargaHoraria { get; set; }
        public string Descricao { get; set; }
        public int Visualizacao { get; set; }

        public Empresa IdEmpresaNavigation { get; set; }
        public Endereco IdEnderecoNavigation { get; set; }
        public ICollection<BeneficioVaga> BeneficioVaga { get; set; }
        public ICollection<Candidatura> Candidatura { get; set; }
        public ICollection<HabilidadeVaga> HabilidadeVaga { get; set; }
        public ICollection<Matching> Matching { get; set; }
    }
}
